queue_processors/queue_processor/queue_processor.py

```diff
@@ -71,12 +72,10 @@ def on_message(self, headers, message):
         self._priority = headers["priority"]
         logger.info("Destination: %s Priority: %s", destination, self._priority)
         # Load the JSON message and header into dictionaries
-        try:
-            self._data_dict = json.loads(message)
-        except ValueError:
-            logger.error("Could not decode message from %s", destination)
-            logger.error(sys.exc_info()[1])
-            return
+        if not isinstance(message, Message):
+            raise ValueError("message received is not of type Message")
+        self.message = message
+
```

Comment:
Here we still want to initalise the message using the Message class, but just pass in the JSON object and allow it not use the json.load message here. The above won't work as AMQ can't pass Message class objects, it has to be JSON